src/pages/__test__/Profile.spec.tsx

```diff
@@ -1,135 +1,141 @@
-    wrapper.find('.user-info button').simulate('click')
-
-    expect(deleteFollowProfileMock).toBeCalledTimes(1)
-  })
-
-  it('should display edit profile instead follow in owner self profile page', async function () {
-    getProfileMock.mockResolvedValue(loggedUser)
-    useRootStateMock.mockReturnValue([{ user: loggedUser }, jest.fn()])
-    const wrapper = shallow(<Profile username="@username" />)
-    await new Promise(r => setImmediate(r))
-    wrapper.update()
-
-    expect(wrapper.find('.user-info .action-btn').props().href).toBe('/settings')
-  })
-})
+  it('should display Follow when user is not following', async () => {
+    const user = generateProfile();
+    user.following = false;
+    getProfileMock.mockResolvedValue(user);
+    const wrapper = shallow(<Profile username={`@${user.username}`} />);
+    await new Promise((r) => setImmediate(r));
+    wrapper.update();
+
+    expect(wrapper.find('.user-info button').text()).toContain('Follow');
+  });
+
+  it('should display Unfollow when user is following', async () => {
+    const user = generateProfile();
+    user.following = true;
+    getProfileMock.mockResolvedValue(user);
+    const wrapper = shallow(<Profile username={`@${user.username}`} />);
+    await new Promise((r) => setImmediate(r));
+    wrapper.update();
+
+    expect(wrapper.find('.user-info button').text()).toContain('Unfollow');
+  });
+
+  it('should send follow request when Follow button clicked', async () => {
+    const user = generateProfile();
+    getProfileMock.mockResolvedValue({ ...user, following: false });
+    const wrapper = shallow(<Profile username={`@${user.username}`} />);
+    await new Promise((r) => setImmediate(r));
+
+    const postFollowProfileMock = postFollowProfile as jest.Mock<Promise<Profile>>;
+    postFollowProfileMock.mockImplementation();
+    wrapper.find('.user-info button').simulate('click');
+
+    expect(postFollowProfileMock).toBeCalledTimes(1);
+  });
+
+  it('should send unfollow request when Unfollow button clicked', async () => {
+    const user = generateProfile();
+    getProfileMock.mockResolvedValue({ ...user, following: true });
+    const wrapper = shallow(<Profile username={`@${user.username}`} />);
+    await new Promise((r) => setImmediate(r));
+
+    const deleteFollowProfileMock = deleteFollowProfile as jest.Mock<Promise<Profile>>;
+    deleteFollowProfileMock.mockImplementation();
+    wrapper.find('.user-info button').simulate('click');
+
+    expect(deleteFollowProfileMock).toBeCalledTimes(1);
+  });
+
+  it('should display edit profile instead follow in owner self profile page', async () => {
+    getProfileMock.mockResolvedValue(loggedUser);
+    useRootStateMock.mockReturnValue([{ user: loggedUser }, jest.fn()]);
+    const wrapper = shallow(<Profile username="@username" />);
+    await new Promise((r) => setImmediate(r));
+    wrapper.update();
+
+    expect(wrapper.find('.user-info .action-btn').props().href).toBe('/settings');
```

Comment:
when I change it to 

```js
  it('should display edit profile instead follow in owner self profile page', async () => {
    getProfileMock.mockResolvedValue(loggedUser);
    useRootStateMock.mockReturnValue([{ user: loggedUser }, jest.fn()]);
    const wrapper = render(<Profile username="@username" />);
    await new Promise((r) => setImmediate(r));

    expect(wrapper.find('.user-info .action-btn').prop('href')).toBe('/settings');
  });
```

it will pass, I guess it might be because enzyme doesn’t know class

Reply:
using `render` instead of `shallow`

Reply:
`enzyme-adapter-preact-pure` should be handling that, but you're right. I'll bring this up with the guy who manages our Enzyme stuff, see if he knows what's going on.

Reply:
Switching to mount worked: https://github.com/preactjs/enzyme-adapter-preact-pure#property-names